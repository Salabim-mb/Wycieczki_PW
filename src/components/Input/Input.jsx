import React from 'react';
import { TextField } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

const Input = ({ value, handleChange, error, id, ...rest }) => (
	<>
		<TextField name={id} variant="outlined" value={value} onChange={handleChange} {...rest} />
		{error && <Alert severity="error">{error}</Alert>}
	</>
);

export default Input;
