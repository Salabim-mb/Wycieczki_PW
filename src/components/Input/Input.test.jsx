import React from 'react';
import { render } from '@testing-library/react';
import Input from './Input';

describe('Input', () => {
	const props = {
		value: '',
		handleChange: jest.fn(),
		error: '',
		id: 'input',
		label: 'Input',
	};

	it('should match snapshot', () => {
		const { container } = render(<Input {...props} />);

		expect(container).toMatchSnapshot();
	});
});
