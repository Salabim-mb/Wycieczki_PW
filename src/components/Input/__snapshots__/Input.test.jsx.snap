// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
    rest="[object Object]"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      Input
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        name="input"
        placeholder=""
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            Input
          </span>
        </legend>
      </fieldset>
    </div>
    
  </div>
</div>
`;
